packages/captp/lib/captp.js

```diff
@@ -1,15 +1,31 @@
 // This logic was mostly lifted from @agoric/swingset-vat liveSlots.js
 // Defects in it are mfig's fault.
-import { Remotable, makeMarshal, QCLASS } from '@agoric/marshal';
-import harden from '@agoric/harden';
-import Nat from '@agoric/nat';
-import { HandledPromise, E } from '@agoric/eventual-send';
+import {
+  Remotable as defaultRemotable,
+  makeMarshal,
+  QCLASS,
+} from '@agoric/marshal';
+import defaultHarden from '@agoric/harden';
+import { HandledPromise as defaultHandledPromise } from '@agoric/eventual-send';
 import { isPromise } from '@agoric/produce-promise';
 
-export { E, HandledPromise, Nat, harden };
-
-export function makeCapTP(ourId, send, bootstrapObj = undefined) {
+export function makeCapTP(
+  ourId,
+  rawSend,
+  bootstrapObj = undefined,
+  {
+    Remotable = defaultRemotable,
+    HandledPromise = defaultHandledPromise,
+    harden = defaultHarden,
+  } = {},
```

Comment:
Recently I found myself needing to express nested named parameter defaults and failed to invent this pattern. Nice!

However, the idea of overriding these three seems scary. Why needed? Is this preparing for three party handoff? I don't get it.

Reply:
It looks like allowing tests to override the internals, or to let a specific environment provide the `HandledPromise` that it should use.

Would it be safer to require the caller to provide `makeCapTP` with `HandledPromise`/etc all the time? I don't have a clear idea of what code needs to create a new capTP instance, and whether it would be an imposition.